reconcile/utils/terrascript_aws_client.py

```diff
@@ -5148,6 +5148,73 @@ def _get_alb_rule_condition_value(condition):
             raise KeyError(f"unknown alb rule condition type {condition_type}")
         return {condition_type_key: {"values": condition[condition_type_key]}}
 
+    @staticmethod
+    def _get_elb_account_id(region: str) -> str:
+        """This mapping comes from
+        https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html#access-log-create-bucket
+        """
+        elb_account_map = {
+            "us-east-1": "127311923021",
+            "us-east-2": "033677994240",
+            "us-west-1": "027434742980",
+            "us-west-2": "797873946194",
+            "af-south-1": "098369216593",
+            "ap-east-1": "754344448648",
+            "ap-southeast-3": "589379963580",
+            "ap-south-1": "718504428378",
+            "ap-northeast-3": "383597477331",
+            "ap-northeast-2": "600734575887",
+            "ap-southeast-1": "114774131450",
+            "ap-southeast-2": "783225319266",
+            "ap-northeast-1": "582318560864",
+            "ca-central-1": "985666609251",
+            "eu-central-1": "054676820928",
+            "eu-west-1": "156460612806",
+            "eu-west-2": "652711504416",
+            "eu-south-1": "635631232127",
+            "eu-west-3": "009996457667",
+            "eu-north-1": "897822967062",
+            "me-south-1": "076674570225",
+            "sa-east-1": "507241528517",
+            "us-gov-west-1": "048591011584",
+            "us-gov-east-1": "190560391635",
+        }
+
+        return elb_account_map[region]
+
+    @staticmethod
+    def _get_principal_for_s3_bucket_policy(
+        region: str, elb_account_id: str | None
+    ) -> Mapping[str, str]:
+        if region in {
+            "us-east-1",
+            "us-east-2",
+            "us-west-1",
+            "us-west-2",
+            "af-south-1",
+            "ap-east-1",
+            "ap-southeast-3",
+            "ap-south-1",
+            "ap-northeast-3",
+            "ap-northeast-2",
+            "ap-southeast-1",
+            "ap-southeast-2",
+            "ap-northeast-1",
+            "ca-central-1",
+            "eu-central-1",
+            "eu-west-1",
+            "eu-west-2",
+            "eu-south-1",
+            "eu-west-3",
+            "eu-north-1",
+            "me-south-1",
+            "sa-east-1",
+        }:
+            return {"AWS": f"arn:aws:iam::{elb_account_id}:root"}
+        if region in {"us-gov-west-1", "us-gov-east-1"}:
+            return {"AWS": f"arn:aws-us-gov:iam::{elb_account_id}:root"}
+        return {"Service": "logdelivery.elasticloadbalancing.amazonaws.com"}
```

Comment:
to avoid duplicate region names in 2 places, can extract mapping to 2 constants, then just get principal by region

```python
AWS_ELB_ACCOUNT_IDS = {
  "us-east-1": "127311923021",
  ...
}

AWS_US_GOV_ELB_ACCOUNT_IDS = {
  "us-gov-west-1": "048591011584",
  "us-gov-east-1": "190560391635",
}

    @staticmethod
    def _get_principal_for_s3_bucket_policy(region: str) -> Mapping[str, str]:
        if region in AWS_ELB_ACCOUNT_IDS:
            return {"AWS": f"arn:aws:iam::{AWS_ELB_ACCOUNT_IDS[region]}:root"}
        if region in AWS_US_GOV_ELB_ACCOUNT_IDS:
            return {"AWS": f"arn:aws-us-gov:iam::{AWS_US_GOV_ELB_ACCOUNT_IDS[region]}:root"}
        return {"Service": "logdelivery.elasticloadbalancing.amazonaws.com"}
```

Reply:
Done, thanks for the tip.

Reply:
@esron please check again and test it, you are using a non existing method `_get_elb_account_id`, please delete `elb_account_id = self._get_elb_account_id(region)` and do everything in `principal = self._get_principal_for_s3_bucket_policy(region)`. I'll revert this change for now